debian/openscad-testrun.1
.TH OPENSCAD-TESTRUN 1 "2012-02-16"
.\" Please adjust this date whenever revising the manpage.
.SH NAME
openscad-testrun \- set up and run the OpenSCAD test suite
.SH SYNOPSIS
.B openscad-testrun
.RI [ options ]
.SH DESCRIPTION
This manual page documents briefly the \fBopenscad-testrun\fP command.
.PP
\fBopenscad-testrun\fP is a script that sets up a directory in which the
OpenSCAD test suite (implemented in ctest) can be run by creating symlinks to
the system locations of the input data, and runs it. The created directory is
not removed, but will contain the test results in addition to the symlinks.

It is required as the test suite in its original form expects all the test
input and output data in relative locations, and the typical user has no write
access to where the data resides. Future changes to the test suite might make
it more flexible, removing the need for this script.
.SH OPTIONS

.TP
\fB\-d\fP \fIdirectory\fP, \fB\-\-directory\fP \fIdirectory\fP
Set up the tests in a directory called \fIdirectory\fP. By default, this is
generated from the current date and time like
\fBopenscad-test-2012-02-15_13:37\fP.
.TP
\fB\-n\fP, \fB\-\-dry\-run\fP
Do not run the test suite, just set it up.
.PP
All additional arguments are passed on to ctest.
.SH AUTHOR
The OpenSCAD test suite was written by Clifford Wolf, Marius Kintel, and
others. The \fBopenscad-testrun\fP script was written by chrysn.
.PP
This manual page was written by chrysn <chrysn@fsfe.org>,
for the Debian project (and may be used by others).


debian/changelog
openscad (2011.12-2) unstable; urgency=low

  * Add a patch for tests/CMakeLists.txt so it works with -Wl,--as-needed too
  * Add a patch to include GL/glu where needed (required since Ubuntu precise)
  * Depend on cmake (ctest) for openscad-testing

 -- Christian M. Amsüss <chrysn@fsfe.org>  Wed, 22 Feb 2012 17:35:12 +0100

openscad (2011.12-1) unstable; urgency=low

  * New upstream release
    - New functions: version, version_num, len
    - Enhanced functionaltiy: hull(), color()
    - Unified import statement, including OFF file format
    - Enhanced string handling: indexing, comparison
    - standard shortcuts for save and reload
    - Bugfixes
  * Dropped man page (in upstream now)
  * Ship test suite as -testing and -testing-data (introduces new
    dependencies)
  * Adapted GLEW dependency to avoid something similar as bug #658551
  * License issue around OGL_helper.h resolved by CGAL moving to GPL-3+

 -- Christian M. Amsüss <chrysn@fsfe.org>  Wed, 15 Feb 2012 13:34:04 +0100

openscad (2011.11~.03-git3bc03713-1) experimental; urgency=low

  * More precise eigen dependency
  * Changed version format to better reflect how upstream calls unreleased
    builds
  * Imported changes from ustream git
    - string comparison
    - version()/version_num() functions
    - new PI constant
    - standard shortcuts for save and reload
    - bugfixes
  * Dropped openscad.pro patch (now takes PREFIX=/usr)
  * Copyright file adaptions to new files
  * Use debian/clean instead of override_dh_auto_clean
  * Run test suite after building
    - new build dependencies: python, cmake
  * Updated man page

 -- Christian M. Amsüss <chrysn@fsfe.org>  Fri, 04 Nov 2011 16:25:00 +0100

openscad (2011.09~git20110929-1) experimental; urgency=low

  * Imported changes from upstream git
    - OFF file format support
    - universal import statement (deprecating some others)
    - support for alpha channel and named colors
    - more flexible if statement
    - much refactoring ("visitor branch")
    - bugfixes
  * Copyright clarifications
  * Suggest editors / viewers for typical import and export formats
  * Recommend openscad-mcad (standard library)
  * Added README.source concerning how to obtain the source package

 -- Christian M. Amsüss <chrysn@fsfe.org>  Fri, 30 Sep 2011 03:08:01 +0200

openscad (2011.06-2) unstable; urgency=low

  * Adapt libglew dependency to 1.6 | anything
  * Updated watchfile to github hosting
  * Updated copyright to current DEP5 version

 -- Christian M. Amsüss <chrysn@fsfe.org>  Sat, 24 Sep 2011 03:29:41 +0200

openscad (2011.06-1) unstable; urgency=low

  * New upstream release
    - Added image export
    - Fixed DXF export regression
    - Fixed cylinder orientation regression

 -- Christian M. Amsüss <chrysn@fsfe.org>  Tue, 14 Jun 2011 19:48:07 +0200

openscad (2011.04-1) unstable; urgency=low

  * New upstream release 
    - Added hull() for convex hulls (2D object only)
    - minkowski() now supports 2D objects
    - DXF output uses LWPOLYLINE instead of LINE
  * Change Standards-Version to 3.9.2 (no changes)
  * Add proper header to patch file

 -- Christian M. Amsüss <chrysn@fsfe.org>  Wed, 27 Apr 2011 19:56:29 +0200

openscad (2011.01~git20110204-2) unstable; urgency=low

  * Removing versioned dependency on libgmp dependency according to upstream
  * Removing versioned dependency on libmpfr (building works w/o as well,
    especially concerning Ubuntu lucid)

 -- Christian M. Amsüss <chrysn@fsfe.org>  Sun, 03 Apr 2011 14:35:52 +0200

openscad (2011.01~git20110204-1) unstable; urgency=low

  * Import changes from upstream git
    - Added rands() function
    - Added support for string escapes
    - Enhancements in nested includes, number parsing, and DXF output
    - Bugfixes
  * New library dependencies
  * Copyright updates
  * Leave installing examples to the upstream installer
  * Keep openscad.desktop in debian/

 -- Christian M. Amsüss <chrysn@fsfe.org>  Mon, 21 Feb 2011 22:40:53 +0100

openscad (2010.06~svn568-1) unstable; urgency=low

  * Import bugfixes from upstream SVN
    - Enhance DXF export for Inkscape import
    - Better CGAL exception handling
    - Offer to save when creating new file
    - New sign() function
  * Change Standards-Version to 3.9.1 (no changes)
  * Removed examples patch, symlinking instead

 -- Christian M. Amsüss <chrysn@fsfe.org>  Mon, 13 Sep 2010 21:19:16 +0200

openscad (2010.06~svn560-1) unstable; urgency=low

  * Import bugfixes from upstream SVN
    - Accept 3D parameters for 2D functions
    - Fix locale dependent export problems
  * New man page for openscad

 -- Christian M. Amsüss <chrysn@fsfe.org>  Sat, 26 Jun 2010 18:28:12 +0200

openscad (2010.05-2) unstable; urgency=low

  * New openscad.desktop
  * Create openscad.png with aspect ratio appropriate for icons
    - new build dependency on imagemagick
  * Install TODO.txt

 -- Christian M. Amsüss <chrysn@fsfe.org>  Sat, 29 May 2010 00:54:19 +0200

openscad (2010.05-1) unstable; urgency=low

  * New upstream release:
    - more math functions
    - support for minkowski sums
    - new include syntax
    - bugfixes
  * Ship RELEASE_NOTES file
  * Don't compress example files
  * Patch openscad.cc to look for examples at the right place

 -- Christian M. Amsüss <chrysn@fsfe.org>  Tue, 18 May 2010 22:34:25 +0200

openscad (2010.02-2) unstable; urgency=low

  * Better build dependencies
  * Added Vcs fields 

 -- Christian M. Amsüss <chrysn@fsfe.org>  Tue, 02 Mar 2010 09:35:09 +0100

openscad (2010.02-1) unstable; urgency=low

  * Initial packaging

 -- Christian M. Amsüss <chrysn@fsfe.org>  Mon, 22 Feb 2010 21:30:49 +0100


debian/openscad-testrun.py
#!/usr/bin/env python

import argparse
import os
import datetime
from os.path import join

TESTPROGRAMS = "/usr/lib/openscad/testprograms"
EXAMPLES = "/usr/share/openscad/examples"
TESTDATA = "/usr/share/openscad/testdata"
REGRESSION = "/usr/share/openscad/regression"

def main():
    p = argparse.ArgumentParser(description="Set up and run the OpenSCAD test suite")
    p.add_argument("-d", "--directory", help="Where to set up the files")
    p.add_argument("-n", "--no-run", help="Just set up the files, don't run", action="store_true")
    p.add_argument("additional", nargs="*", help="remaining arguments will be passed to ctest")

    args = p.parse_args()

    dirname = args.directory or "openscad-test-%s"%datetime.datetime.now().strftime("%Y-%m-%d_%H:%M")
    builddir = join(dirname, "build")

    print "Creating test infrastructure in", dirname

    os.mkdir(dirname)
    os.mkdir(builddir)
    os.symlink(EXAMPLES, join(dirname, "examples"))
    os.symlink(TESTDATA, join(dirname, "testdata"))
    os.symlink(REGRESSION, join(builddir, "regression"))

    for f in os.listdir(TESTPROGRAMS):
        os.symlink(join(TESTPROGRAMS, f), join(builddir, f))

    if args.no_run:
        print "Skipping test run"
    else:
        os.chdir(builddir)
        os.execvp("ctest", ["ctest"] + args.additional)

if __name__ == "__main__":
    main()
